swap/src/protocol/alice/event_loop.rs

```diff
@@ -195,19 +195,40 @@ where
                             }
                         }
                         SwarmEvent::Behaviour(OutEvent::EncryptedSignatureReceived{ msg, channel, peer }) => {
-                            let sender = match self.recv_encrypted_signature.remove(&msg.swap_id) {
+                            let swap_id = msg.swap_id;
+                            let swap_peer = self.db.get_peer_id(swap_id);
+
+                            // Ensure that an incoming encryption signature is sent by the peer-id associated with the swap
+                            match swap_peer {
```

Comment:
I think this would be easier to read if we use two `if let` statements because the current "happy-path" is hidden within the first match arm when the `if` condition **doesn't** evaluate to true. That is quite hard to understand IMO.

Reply:
I don't really understand how using two `if let` would make this section better. Could you give me some small example of what you have in mind. 
This is an early exit on error scenario - I don't understand why the happy path is hidden. There are other early exit scenarios defined in the block, the happy path comes afterwards.

Reply:
This `match` is a statement, not an expression, meaning the happy path goes **through** this statement instead of around it like we do in other places. Because it is not an expression, I have to evaluate the whole code before I know, what branches there are and under which conditions I am actually continuing with the code.

In particular, the happy path is hidden within the `Ok` branch if `swap_peer != peer` is `false` and this is only the happy path because all other branches use `continue`. In theory, there could be more paths through this statement. This makes it fairly hard to understand.

Compare that to:

```rust
let swap_peer = match swap_peer {
	Ok(swap_peer) => swap_peer,
	Err(_) => {
		tracing::warn!("Ignoring encrypted signature for unknown swap {} from {}", swap_id, peer);
        continue;
	}
};

if swap_peer != peer {
    tracing::warn!(
        %swap_id,
        "Ignoring malicious encrypted signature from {}, expected to receive it from {}",
        peer,
        swap_peer);
    continue;
}
```

Reply:
Ah, sorry, that was a misunderstanding then. Sure, splitting the `match` statement into two distinct parts is a good idea. I think I was confused by 

> two if let statements

because I could not see how I could make that happen :)
Done now.

Reply:
Yes, initially I thought we need two `if let`s. Don't take what I say too literally :)